src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -5391,7 +5391,12 @@ void QueryAnalyzer::initializeTableExpressionColumns(const QueryTreeNodePtr & ta
     {
         const auto & storage_snapshot = table_node ? table_node->getStorageSnapshot() : table_function_node->getStorageSnapshot();
 
-        auto column_names_and_types = storage_snapshot->getColumns(GetColumnsOptions(GetColumnsOptions::All).withSubcolumns().withVirtuals());
+        auto column_names_and_types = storage_snapshot->getColumns(
```

Comment:
Please modify it like it is done here https://github.com/ClickHouse/ClickHouse/pull/45461/files#diff-377c0cb9048fe3fe24b9501a4d6fdadde00a04c75a162f705de66cadeb654d9cR5630. We should not use `withSubcolumns` is storage does not support it.

Reply:
Maybe just close this PR then and move tests to your PR?

Reply:
Let's merge this one, I will just use them after merge.